site/security_providers.md

```diff
@@ -0,0 +1,109 @@
+                            ControlFlag: SUFFICIENT
+                            PropagateCauseForLoginException: true
+                            EnableGroupMembershipLookupHierarchyCaching: true
+                            Host: myldap.example.com
+                            Port: 389
+                            UserObjectClass: person
+                            GroupHierarchyCacheTTL: 600
+                            SSLEnabled: true
+                            UserNameAttribute: cn
+                            Principal: 'cn=foo,ou=users,dc=example,dc=com'
+                            UserBaseDn: 'OU=Users,DC=example,DC=com'
+                            UserSearchScope: subtree
+                            UserFromNameFilter: '(&(cn=%u)(objectclass=person))'
+                            AllUsersFilter: '(memberOf=CN=foo,OU=mygroups,DC=example,DC=com)'
+                            GroupBaseDN: 'OU=mygroups,DC=example,DC=com'
+                            AllGroupsFilter: '(&(foo)(objectclass=group))'
+                            StaticGroupObjectClass: group
+                            StaticMemberDNAttribute: cn
+                            StaticGroupDNsfromMemberDNFilter: '(&(member=%M)(objectclass=group))'
+                            DynamicGroupObjectClass: group
+                            DynamicGroupNameAttribute: cn
+                            UseRetrievedUserNameAsPrincipal: true
+                            KeepAliveEnabled: true
+                            GuidAttribute: uuid
+                    My LDAP IdentityAsserter:
+                        LDAPX509IdentityAsserter:
+                            ActiveType: AuthenticatedUser
+                            Host: myldap.example.com
+                            Port: 389
+                            SSLEnabled: true
+                    DefaultAuthenticator:
+                        DefaultAuthenticator:
+                            ControlFlag: SUFFICIENT
+                    DefaultIdentityAsserter:
+                        DefaultIdentityAsserter:
+
+```
+
+#### Custom Security Providers
+
+**NOTE:** Creating and updating domains with custom security providers is limited to WebLogic version 12.1.2 and newer.
+
+Prior to using this tooling to create or update a domain with a custom security provider, there are several prerequisites.  First, WebLogic Server requires the custom MBean JAR to be in the Oracle Home directory before it can be configured, WLSERVER/server/lib/mbeantypes.  Second, WebLogic Scripting Tool, WLST, requires that the schema JAR be placed in the Oracle Home directory before WLST offline can be used to configure it, ORACLEHOME/oracle_common/lib/schematypes.  Generating an MBean JAR documentation can be found in the WebLogic Server [documentation](https://docs.oracle.com/middleware/12213/wls/DEVSP/generate_mbeantype.htm#DEVSP617).  Generating the schema JAR can be done with the prepareCustomProvider script provided in the WebLogic Server installation.
```

Comment:
* WLSERVER/server/lib/mbeantypes -> `WLSERVER/server/lib/mbeantypes` (code font)
* ORACLEHOME/oracle_common/lib/schematypes -> `ORACLEHOME/oracle_common/lib/schematypes` (code font)
* prepareCustomProvider -> `prepareCustomProvider` (code font)

Reply:
corrected in next push, thanks